### PR TITLE
Symfony config

### PR DESCRIPTION
**Pull request by JerryVerhoef**
WIP - please don't merge yet

https://github.com/qandidate-labs/qandidate-toggle-bundle/issues/21

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function getConfigTreeBuilder()
         $rootNode
             ->children()
                 ->enumNode('persistence')
-                    ->values(array('in_memory', 'redis', 'factory'))
+                    ->values(array('in_memory', 'redis', 'factory', 'symfony'))
```
Comment (asm89) [inline DependencyInjection/Configuration.php]: nit: should we call this `configfile`? Or something else that is not "symfony"?

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -53,7 +52,26 @@ public function getConfigTreeBuilder()
                 ->scalarNode('redis_client')
                     ->defaultNull()
                 ->end()
-
+                ->arrayNode('toggles')
```
Comment (asm89) [inline DependencyInjection/Configuration.php]: Let's add some validation that you can only set this if `persistence` enables this option.

File: Factory/SymfonyCollection.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Qandidate\Bundle\ToggleBundle\Factory;
+
+use Qandidate\Toggle\Serializer\InMemoryCollectionSerializer;
+
+/**
+ * Class SymfonyCollection
+ * @package Qandidate\Bundle\ToggleBundle\Factory
+ */
+class SymfonyCollection
```
Comment (asm89) [inline Factory/SymfonyCollection.php]: `InMemoryCollectionFactory`?

File: QandidateToggleBundle.php [hunk 4]
```diff
@@ -11,8 +11,11 @@
 
 namespace Qandidate\Bundle\ToggleBundle;
 
+use Qandidate\Bundle\ToggleBundle\DependencyInjection\QandidateToggleExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
 class QandidateToggleBundle extends Bundle
 {
+
```
Comment (asm89) [inline QandidateToggleBundle.php]: revert please

**Comment by asm89** [comment]
@JerryVerhoef The overall idea of this PR looks good to me. :+1: 

If you're up for it, it'd be great if you can split the pull request in two parts:
- `composer.json` bump to use `v1.0` (separate PR)
- new feature addition

**Comment by asm89** [comment]
@JerryVerhoef @fabian-markaban Any news here?

**Comment by fabian-markaban** [comment]
@asm89 Hi Alexander, we are working on a final version and will push this before the end of the week. Greetings!

**Comment by asm89** [comment]
@fabian-markaban @JerryVerhoef  Still interested in pushing this out?

**Comment by asm89** [comment]
For those interesting. See here https://github.com/qandidate-labs/qandidate-toggle-bundle/pull/33.

**Pull request closed without merging**